controllers/component_controller.go

```diff
@@ -47,10 +60,197 @@ type ComponentReconciler struct {
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to update the HAS Application Devfile model %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, nil
+				}
+
+				yamlHASCompData, err := yaml.Marshal(hasCompDevfileData)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to marshall the Component devfile, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				hasComponent.Status.Devfile = string(yamlHASCompData)
+
+				// Update the HASApp CR with the new devfile
+				yamlHASAppData, err := yaml.Marshal(hasAppDevfileData)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to marshall the Application devfile, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+				hasApplication.Status.Devfile = string(yamlHASAppData)
+				err = r.Status().Update(ctx, &hasApplication)
+				if err != nil {
+					log.Error(err, "Unable to update Application")
+					// if we're unable to update the Application CR, then  we need to err out
+					// since we need to save a reference of the Component in Application
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, nil)
+
+				log.Info(fmt.Sprintf("Updating the labels for HAS Component %v", req.NamespacedName))
+				hasComponentLabels := make(map[string]string)
+				hasComponentLabels["appstudio.has/application"] = hasComponent.Spec.Application
+				hasComponentLabels["appstudio.has/component"] = hasComponent.Spec.ComponentName
+				hasComponent.SetLabels(hasComponentLabels)
+				err = r.Client.Update(ctx, &hasComponent)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to update Component with the required labels %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+			} else {
+				log.Error(err, fmt.Sprintf("Application devfile model is empty. Before creating a Component, an instance of Application should be created, exiting reconcile loop %v", req.NamespacedName))
+				err := fmt.Errorf("application devfile model is empty. Before creating a Component, an instance of Application should be created")
+				r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+				return ctrl.Result{}, err
+			}
+
+		} else if source.ImageSource != nil && source.ImageSource.ContainerImage != "" {
+			log.Info(fmt.Sprintf("container image is not supported at the moment, please use github links for adding a component to an application for %v", req.NamespacedName))
+		}
+	} else {
+		// If the model already exists, see if fields have been updated
+		log.Info(fmt.Sprintf("Checking if the HAS Component has been updated %v", req.NamespacedName))
```

Comment:
```suggestion
		log.Info(fmt.Sprintf("Checking if the Component has been updated %v", req.NamespacedName))
```

🙂  

There's a few other instances too